### PR TITLE
fix(core): router

### PR DESCRIPTION
**Pull request by AinaVendrell**
## Description
Fix a bug in getting child routes that was added in https://github.com/hubtype/botonic/pull/903. When the bot was receiving a payload like : payload: '__PATH __PAYLOAD_2' and also had a lastRoute the lastRoute wasn't being checked and it wasn't able to find the route.

## Testing

The pull request...

- has unit tests

### CODE REVIEW

**Review by vanbasten17 — APPROVED** [review]
LGTM 👍 we would be able to do more changes when we have improved the tests a little more

File: packages/botonic-core/src/router.test.js [hunk 1]
```diff
@@ -225,6 +225,14 @@ describe('Process input (v<0.9)', () => {
       'Flow1.2'
     )
   })
+  test('old protocol: path payload input, 2nd level route with lastRoutePath', () => {
```
Comment (dpinol) [inline packages/botonic-core/src/router.test.js:228]: please extend the PR description to explain what was the bug.
Also, I don't understand where is lastRoutePath used here

**Review by dpinol — CHANGES_REQUESTED** [review]

File: packages/botonic-core/src/router.js [hunk 2]
```diff
@@ -149,34 +150,24 @@ export class Router {
...
-          if (previousValue) currentValue = previousValue.concat('/', value)
-          const match = this.matchRoute(r, key, currentValue, input, session)
-          if (!match && r.childRoutes && !childRoute) {
-            childRoute = this.getRoute(
-              input,
-              r.childRoutes,
-              session,
-              currentValue
-            )
-          }
```
Comment (asastre) [inline packages/botonic-core/src/router.js:174]: Are we losing something removing this code?

Comment (AinaVendrell) [reply packages/botonic-core/src/router.js:174]: I added this in the old PR because I thought it was necessary but it was not.

**Review by asastre — APPROVED** [review]

**Pull request merged**